### PR TITLE
Checkpoint fixes 

### PR DESCRIPTION
**Pull request by bsuieric**
Each commit is associated with a ticket related to checkpointing. These are some of the fixes applied by Pawel

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcher.scala [hunk 1]
```diff
@@ -40,7 +40,7 @@ class BlockFetcher(
     val syncConfig: SyncConfig,
     val blockValidator: BlockValidator
 ) extends Actor
-    with ActorLogging {
+    with Logger {
```
Comment (pslaski) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcher.scala]: `ActorLogging` should stay for all actors

File: src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala [hunk 2]
```diff
@@ -37,7 +37,7 @@ class BlockImporter(
     checkpointBlockGenerator: CheckpointBlockGenerator,
     supervisor: ActorRef
 ) extends Actor
-    with ActorLogging {
```
Comment (pslaski) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala]: same here

File: src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala [hunk 3]
```diff
@@ -249,7 +247,7 @@ class BlockImporter(
       informFetcherOnFail: Boolean,
       internally: Boolean
   ): ImportFn = {
-    def doLog(entry: ImportMessages.LogEntry): Unit = log.log(entry._1, entry._2)
+    def doLog(entry: ImportMessages.LogEntry): Unit = log.info(entry._2)//log.log(entry._1, entry._2)
```
Comment (pslaski) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala]: `log(entry._1, entry._2)` should stay here

**Review by pslaski — CHANGES_REQUESTED** [review]

**Review by pslaski — CHANGES_REQUESTED** [review]
Also, run `scalafmt` :)

File: src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcher.scala (hunk 1 above)
Comment (bsuieric) [reply src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcher.scala]: will revert the commit

**Review by pslaski — APPROVED** [review]

File: src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcherState.scala [hunk 4]
```diff
@@ -123,11 +123,18 @@ case class BlockFetcherState(
...
         .ensure("Given headers do not form a chain with already stored ones")(headers =>
           (waitingHeaders.lastOption, headers.headOption).mapN(_ isParentOf _).getOrElse(true)
         )
     }
 
+  private def checkConsistencyWithReadyBlocks(headers: Seq[BlockHeader]): Boolean = {
+    if (waitingHeaders.isEmpty) {
+      (readyBlocks.lastOption.map(_.header), headers.headOption).mapN(_ isParentOf _).getOrElse(true)
+    } else true
+  }
```
Comment (AnastasiiaL) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcherState.scala]:
```suggestion
 private def checkConsistencyWithReadyBlocks(headers: Seq[BlockHeader]): Boolean = {
    (readyBlocks, headers) match {
      case (_ :+ last, head +: _) if waitingHeaders.isEmpty  => last.header isParentOf head
      case _ => true
    }
```

**Review by AnastasiiaL — APPROVED** [review]

**Pull request merged**